extensions/barrage/src/main/java/io/deephaven/extensions/barrage/chunk/ChunkReader.java

```diff
@@ -18,7 +20,56 @@
 /**
  * Consumes Flight/Barrage streams and transforms them into WritableChunks.
  */
-public interface ChunkReader {
+public interface ChunkReader<ReadChunkType extends WritableChunk<Values>> {
```

Comment:
We need these to return the Chunks to support usages such as `LongChunkReader#transformTo`. Since ownership is also being passed, it is best to advertise that they're `WritableChunk`s.